src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -388,7 +388,32 @@ public static bool ShouldSkipCompilation(MethodDesc methodNeedingCode)
             return false;
         }
 
-        public void CompileMethod(MethodWithGCInfo methodCodeNodeNeedingCode)
+        private bool FunctionJustThrows(MethodIL ilBody)
+        {
+            try
+            {
+                if (ilBody.GetExceptionRegions().Length != 0)
+                    return false;
+
+                ILReader reader = new ILReader(ilBody.GetILBytes());
+
+                while (reader.HasNext)
+                {
+                    var ilOpcode = reader.ReadILOpcode();
+                    if (ilOpcode == ILOpcode.throw_)
+                        return true;
+                    if (ilOpcode.IsBranch())
```

Comment:
Nit:

```suggestion
                    if (ilOpcode.IsBranch() || ilOpcode == ILOpcode.switch_)
```

To cover patterns like

```csharp
static int Translate(int val) => val switch
{
    0 => 1,
    1 => 5,
    2 => 6,
    3 => 10,
    _ => throw new Exception()
};
```